### PR TITLE
feat(votes): handle OverrideVoteCast to update delegate votes on override

### PR DESCRIPTION
**Pull request by harryburger**
## 📝 Summary

This pull request introduces support for the `OverrideVoteCast` event, which handles cases where a delegator overrides their delegatee's vote in token voting. The changes ensure that delegatee votes are properly updated or zeroed out when overridden, and that this information is reflected in both the database and API responses. The pull request also includes comprehensive tests for the new logic and updates proposal metrics to exclude overridden (zero-power) votes.

**Support for OverrideVoteCast events:**

- Added the `OverrideVoteCast` event ABI to `TokenVoting` and registered it in the indexer configuration, enabling the system to listen for and process these events. [[1]](diffhunk://#diff-9408633997b8034aee6f541a70551834ff1a160dde6be011b4da1b15567c946eR335-R379) [[2]](diffhunk://#diff-45e850decefd473cc28d1ae4fbef9a9219d2800a5d68ec9e986a27ccd790577dR303-R313) [[3]](diffhunk://#diff-f08c325ce2ac5b04b6c82b61ecee69215676749a377fe7be393f7adf1313bf82R262)
- Implemented the `overrideVoteCast` handler in `ProposalHandler`, which updates the delegatee's vote record with the remaining voting power and marks it as overridden, ensuring idempotency and proper logging.

**Vote model and API enhancements:**

- Added the `VoteOverridden` subdocument to the `Vote` model to track override status, transaction, and block details, and exposed this state in API responses. [[1]](diffhunk://#diff-27b747ada55018ed7e080a29c0d9e0af7de18e38b2930c20cde0699883b669a6R34-R49) [[2]](diffhunk://#diff-27b747ada55018ed7e080a29c0d9e0af7de18e38b2930c20cde0699883b669a6R120-R122) [[3]](diffhunk://#diff-f43b9d97b078302b7432662db3f115fbe115939ada422d07fd3175af71e4faa7R41) [[4]](diffhunk://#diff-56da1d200f822734c9b66206597b6cec9911d1ad396057f270a6f346ea0a95f4R196-R201) [[5]](diffhunk://#diff-27b747ada55018ed7e080a29c0d9e0af7de18e38b2930c20cde0699883b669a6R399)

**Proposal metrics accuracy:**

- Updated proposal metrics calculation to exclude votes with zero voting power (i.e., fully overridden votes) from totals and vote aggregations, ensuring accurate statistics. [[1]](diffhunk://#diff-d5e6442dd8dd08a2c80685d991a17f68a824d3a150c5236c3d56e28031a5a6e6L80-R82) [[2]](diffhunk://#diff-d5e6442dd8dd08a2c80685d991a17f68a824d3a150c5236c3d56e28031a5a6e6L101-R107)

**Testing and validation:**

- Added comprehensive unit tests for the `overrideVoteCast` handler, covering partial/full overrides, idempotency, and error cases.
- Extended vote controller tests to verify that the `voteOverridden` state is correctly returned in both paginated vote lists and individual member vote info. [[1]](diffhunk://#diff-520420dfa89a103e92e744c7ddc56d0637bc0fbe5195a7620fe4fb0484157bc8R197-R220) [[2]](diffhunk://#diff-520420dfa89a103e92e744c7ddc56d0637bc0fbe5195a7620fe4fb0484157bc8R279-R304)

### 🔍 Code Review
- [ ] Self-reviewed all code changes
- [ ] Ask AI/Copilot code review
- [ ] Removed all debug code, console.logs, and dead code
- [ ] Implemented error handling with try/catch blocks where needed

### 🧪 Testing
- [ ] All test suites pass locally
- [ ] Added comprehensive unit tests for new features
- [ ] Included integration tests for end-to-end scenarios
- [ ] Successfully tested on remote server

### 🔒 Security
- [ ] Verified no secrets or credentials are exposed
- [ ] Reviewed for common security vulnerabilities
- [ ] **Verified commit authorship** - Reviewed all commits to ensure they're from team members (check for compromised accounts)

### CODE REVIEW

File: src/models/schema/vote.ts [hunk 1]
```diff
@@ -101,6 +117,9 @@ export default class Vote extends Model {
   @prop({ type: () => VoteCleared, _id: false, default: {} })
   public voteCleared!: VoteCleared
 
+  @prop({ type: () => VoteOverridden, _id: false, default: {} })
+  public voteOverridden!: VoteOverridden
```
[the remark this answers is not in the document]
Comment (milosh86) [reply src/models/schema/vote.ts]: this seems like a valid comment, can we make voteOverridden optional?

File: src/handlers/proposalHandler.ts [hunk 2]
```diff
@@ -420,6 +414,67 @@ export const ProposalHandler = {
...
+      const blockTimestamp = await Web3Helper.getBlockTimestamp(info.blockNumber, info.network)
+
+      const overrideUpdate = {
+        votingPower: parsedEvent.args.delegateeVotingPower.toString(),
+        voteOption: Number(parsedEvent.args.delegateeVoteOption),
+        voteOverridden: {
+          status: true,
+          transactionHash: info.transactionHash,
+          blockNumber: info.blockNumber,
+          blockTimestamp: blockTimestamp || 0,
```
Comment (milosh86) [inline src/handlers/proposalHandler.ts:477]: should override check if blockNumber or blockTimestamp is greater than the existing one to prevent invalid override in the case of out of order events?

**Review by milosh86 — COMMENTED** [review]
a couple of minor details to confirm, looks good.

**Review by milosh86 — APPROVED** [review]
Looks good 👍
